python/mxnet/ndarray/sparse.py

```diff
@@ -727,6 +727,18 @@ def _prepare_src_array(source_array, dtype):
             raise TypeError('values must be array like object')
     return source_array
 
+def _prepare_default_dtype(src_array, dtype):
+    """Prepare the value of dtype if `dtype` is None. If `src_array` is an NDArray, numpy.ndarray
+    or scipy.sparse.csr.csr_matrix, return src_array.dtype. float32 is returned otherwise."""
+    if dtype is None:
+        if isinstance(src_array, (NDArray, np.ndarray)):
+            dtype = src_array.dtype
+        elif spsp and isinstance(src_array, spsp.csr.csr_matrix):
```

Comment:
shouldn't we raise an error if spsp is not available?
Do we currently require scipy when using sparse?

Reply:
scipy is an optional dependency right now. If scipy is not there, spsp is None